android/src/main/java/com/reactlibrary/RNPurchasesModule.java

```diff
@@ -94,16 +97,47 @@ private WritableMap mapForSkuDetails(final SkuDetails detail) {
         map.putString("title", detail.getTitle());
         map.putDouble("price", detail.getPriceAmountMicros() / 1000000d);
         map.putString("price_string", detail.getPrice());
-        String introductoryPriceAmountMicros = detail.getIntroductoryPriceAmountMicros();
-        if (introductoryPriceAmountMicros != null && !introductoryPriceAmountMicros.isEmpty()) {
-            map.putString("intro_price", String.valueOf(Long.parseLong(introductoryPriceAmountMicros) / 1000000d));
+        putIntroPrice(detail, map);
+        map.putString("currency_code", detail.getPriceCurrencyCode());
+        return map;
+    }
+
+    private void putIntroPrice(SkuDetails detail, WritableMap map) {
+        if (detail.getFreeTrialPeriod().isEmpty()) {
+            String introductoryPriceAmountMicros = detail.getIntroductoryPriceAmountMicros();
+            if (introductoryPriceAmountMicros != null && !introductoryPriceAmountMicros.isEmpty()) {
+                map.putString("intro_price", String.valueOf(Long.parseLong(introductoryPriceAmountMicros) / 1000000d));
+            } else {
+                map.putString("intro_price", "");
+            }
+            map.putString("intro_price_string", detail.getIntroductoryPrice());
+            map.putString("intro_price_period", detail.getIntroductoryPricePeriod());
+            if (detail.getIntroductoryPricePeriod() != null && !detail.getIntroductoryPricePeriod().isEmpty()) {
+                RNPurchasesPeriod period = RNPurchasesPeriod.parse(detail.getIntroductoryPricePeriod());
+                if (period.years > 0) {
+                    map.putString("intro_price_period_unit", "YEAR");
+                    map.putString("intro_price_period_number_of_units", "" + period.years);
```

Comment:
Just curious, is there a reason these are all strings / empty-strings? Proper types and nullability would be pretty nice, although I think it would break the API at this point.

Reply:
Yeah, this was done like this at the beginning and now there's no way back without breaking the API. We should probably add TypeScript and proper types